src/enveloping/NonceQueueSelector.ts

```diff
@@ -0,0 +1,66 @@
+import { FeesTable } from './FeeEstimator'
+
+/**
+ * For each transaction that is submitted, chooses which nonce queue will be used, based on existing rules.
+ */
+export class NonceQueueSelector {
+  /**
+   * Returns the recommended gas price for a transaction to be included before the given time, by
+   * applying a determined set of rules.
+   * @param maxTime the timestamp to use for calculation
+   * @param feesTable a fees table to read the gas prices from
+   * @return the recommended gas price
+   */
+  async getQueueGasPrice (maxTime: number, feesTable: FeesTable): Promise<string> {
+    const workerIndex = this.applyDelayRule(maxTime)
+    let gasPrice = '0'
```

Comment:
what would happen if this were to actually be returned?

Reply:
Actually gasPrice = 0 is a valid value, in regtest the gasprice by default is 0, but anyway if that happens the transaction could fail in some scenarios outside regtest.